data/trie/patriciaMerkleTrie_test.go

```diff
@@ -327,6 +327,23 @@ func emptyTrie() data.Trie {
 	return tr
 }
 
+func TestPatriciaMerkleTrie_VerifyProofFromDifferentTrieShouldNotWork(t *testing.T) {
```

Comment:
This unit test does not test for extensionNode, branchNoce and leafNode. There is only one test and cannot test all three cases. 
Please add further tests.

Reply:
Done.

Reply:
ok